livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -1167,4 +1158,43 @@ void Lddc::checkTimeSync(diagnostic_updater::DiagnosticStatusWrapper & stat)
     stat.summary(whole_level, time_sync_dict_.at(whole_level));
   }
 }
+
+void Lddc::checkConnect(diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  if (lidar_count_ == 0) {
+    stat.summary(DiagStatus::WARN, "No LiDARs Connected");
+    return;
+  }
+
+  int whole_level = DiagStatus::OK;
+  std::string error_str = "";
+
+  for (const auto & lidar : lds_->connected_lidars_) {
+    int level = DiagStatus::OK;
+
+    const auto & broadcast_code = lidar.first;
+    const auto & device_info = lidar.second;
+
+    if (device_info == nullptr) {
+      error_str = "LiDAR disconnected";
+      stat.add(broadcast_code, "disconnected");
+      continue;
+    }
+
+    if (device_info->state == kLidarStateInit) {
+      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
+      continue;
+    }
+
+    stat.add(broadcast_code, connect_dict_.at(level));
```

Comment:
It seems `level` is not changed?

Reply:
Please see https://github.com/tier4/livox_ros2_driver/pull/10#issuecomment-849318765